pkg/build/api/v1/types.go

```diff
@@ -163,6 +163,9 @@ type BuildSource struct {
 	// repository.
 	ContextDir string `json:"contextDir,omitempty" description:"specifies sub-directory where the source code for the application exists, allows for sources to be built from a directory other than the root of a repository"`
 
+	// DockerfileName is the name of the Dockerfile that will be used to build the Docker image.
+	DockerfileName string `json:"dockerfileName,omitempty" description:"specifies the name of the Dockerfile to use for building the Docker image"`
```

Comment:
> Dockerfile input needs to be adopted by it (i.e. That strategy needs to use the location)

You mean when writing the content of the Dockerfile to disk, it should be written to dockerfileName instead of the hardcoded "Dockerfile" ?